internal/cli/cli.go

```diff
@@ -134,3 +144,26 @@ func collectionDocuments(serverUrl string, apiKey string, ids []string) error {
 	}
 	return nil
 }
+
+func collectionList(serverUrl string, apiKey string) error {
+	oc := outline.New(serverUrl, &http.Client{}, apiKey)
+	var collections []*outline.Collection
+	err := oc.Collections().List().Do(context.Background(), func(c *outline.Collection, err error) (bool, error) {
+		if err != nil {
+			// We are going to ignore errors for now...
+			return true, nil
+		}
+		collections = append(collections, c)
```

Comment:
Its better to simply print the item as they get received because:
- collecting defeats the purpose of streaming: memory usage is directly proportional to number of items...whereas for streaming memory requirement remains independent.
-  user has to wait until full list gets prepared..depending on number items this can vary greatly...this may confuse user whether cli has hanged or what...
- Also this complicates error handling when say for first 10 items no error but then we have error, now what to print and how to return?

Reply:
Well, I thought in case we ignore the error, we simply "jump" over it (in the http client). So in case we have 10 items, fifth item is broken, it will simply omit this one and continue with the other five. So at the end we would have an slice of 9 collections.

Anyways, I agree with the statement to print out each collection as they receive. That makes the most sense.. 👍